src/session/content/message_row/map.rs

```diff
@@ -0,0 +1,216 @@
+use glib::clone;
+use gtk::prelude::*;
+use gtk::subclass::prelude::*;
+use gtk::{glib, CompositeTemplate};
+use shumate::traits::{LocationExt, MapExt, MarkerExt, MarkerLayerExt};
+
+use crate::session::components::MapMarker;
+
+// The golden ratio
+const PHI: f64 = 1.6180339887;
+const MIN_HEIGHT: i32 = 150;
+const MAX_HEIGHT: i32 = 225;
+const MIN_WIDTH: i32 = (MIN_HEIGHT as f64 * PHI) as i32;
+const MAX_WIDTH: i32 = (MAX_HEIGHT as f64 * PHI) as i32;
+
+pub const CENTER_MARKER_ACTON_NAME: &str = "map.center-marker";
+
+mod imp {
+    use super::*;
+
+    #[derive(Debug, Default, CompositeTemplate)]
+    #[template(string = r#"
+    <interface>
+      <object class="GtkImage" id="marker_image">
+        <property name="css-name">marker</property>
```

Comment:
Css-name should be used just for GtkWidget subclasses, so it's not really a good idea to use it in these cases (except for overrides like we do in the message entry). Prefer using css-classes here.